examples/user_guide/02_Data_Ingestion_with_Intake.ipynb

```diff
@@ -4,10 +4,389 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training = intake.open_csv(['../landsat5_training.csv', '../landsat8_training.csv'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training_df = training.read()\n",
+    "training_df.info()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**NOTE:** The length of the dataframe has increased now that we are loading multiple sets of training data."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This can be more simply expressed as:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training = intake.open_csv('../landsat*_training.csv')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Sometimes, there is data encoded in a file name or path, that causes concatenated data to lose some important context. In this example, we lose the information about which version of landsat the training was done on. To maintain that information you can pass arguments into the csv reader directly like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training = intake.open_csv('../landsat*_training.csv', csv_kwargs=dict(include_path_column=True))\n",
+    "training_df = training.read()\n",
+    "training_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The path information is now included for each file. \n",
+    "\n",
+    "This is helpful, but all we really want is to know the landsat version. To get just that little piece of information we use a python format string to specify our path and declare a new field on our data. That field will get populated based on its value in the path. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training = intake.open_csv('../landsat{version:d}_training.csv')\n",
+    "training_df = training.read()\n",
+    "training_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Using Catalogs\n",
+    "\n",
+    "For more complicated setups, we use the catalog.yml to declare how the data should be loaded. This catalog file lays out how the data should be loaded, defines some metadata, and specifies any patterns in the file path that should be included in the data. Here is an example of a catalog entry:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open('../catalog.yml') as f:\n",
+    "    [print(line.rstrip()) for line in f.readlines()[:16]]"
```

Comment:
Personally, I find using list comprehensions for side-effects (like print) weird. Here I would just use a regular loop in the context manager.